src/sound_tool.c

```diff
@@ -0,0 +1,1285 @@
+    case SOUND_MAIN_THLON:
+    case SOUND_MAIN_THLON_OPED:
+    case SOUND_SUB_PHC:
+        Snd_SetDemoData(scene, no, flag);
+        break;
+    case SOUND_MAIN_DEMO01:
+        Snd_SetDemoData(scene, no, flag);
+        break;
+    default:
+        return FALSE;
+    }
+
+    return TRUE;    
+} 
+
+static void Snd_SetSubData(u8 scene) {
+    int* heap_save_global = GF_SdatGetAttrPtr(SOUND_WORK_HEAP_SAVE_GLOBAL);
+
+    GF_Snd_LoadState(*heap_save_global); 
+
+    GF_Snd_SaveState(GF_SdatGetAttrPtr(SOUND_WORK_HEAP_SAVE_BGM_BANK));   
+
+    Snd_LoadGroupByScene(scene); 
+    GF_Snd_SaveState(GF_SdatGetAttrPtr(SOUND_WORK_HEAP_SAVE_SE));   
+}
+
+static void Snd_SetFieldData(u16 no, int flag) {
+    int temp;
+
+    u8* field_pause_flag = GF_SdatGetAttrPtr(SOUND_WORK_FIELD_PAUSE_FLAG);
+    int* heap_save_global = GF_SdatGetAttrPtr(SOUND_WORK_HEAP_SAVE_GLOBAL);
+    u16* zone_bgm = GF_SdatGetAttrPtr(SOUND_WORK_ZONE_BGM);
+
+    temp = GF_SndPlayerGetSeqNo(GF_GetSoundHandle(SOUND_WORK_WAVEOUT_HANDLE_NORMAL));
+
+    u16 player_seq_no = (u16) temp;
+
+    if (temp > SEQ_GS_P_START) {
+        player_seq_no = GBSounds_GetDSSeqNoByGBSeqNo(player_seq_no);
+    }
+
+    if (*field_pause_flag == 0) {
+        if (player_seq_no == no) {
+            if (Snd_GetNextBGM() != SEQ_GS_BICYCLE) { 
+                return;
+            }
+        }
+    }
+
+    Snd_SetBankFlag(1);  
+
+    if (_021D05E8.unkc == 0 || unk4 == 0) {
+        sub_02005FD8();
+        unk4 = 1;
+    }                                    
+
+    if (player_seq_no != no) {
+        SndPause(PLAYER_FIELD, FALSE);      
+        sub_02005FA0();
+    }
+
+    if (*field_pause_flag == 1) {
+
+        GF_Snd_LoadState(Snd_GetSaveHeap(SOUND_SAVE_HEAP_BGM_BANK));
+
+        Snd_LoadGroupByScene(SOUND_MAIN_FIELD);                     
+        GF_Snd_SaveState(GF_SdatGetAttrPtr(SOUND_WORK_HEAP_SAVE_SE));  
+
+        if (player_seq_no != no) {
+            SndPause(PLAYER_FIELD, FALSE);              
+        }
+
+        Snd_SetFieldDataWithoutDelay(no, temp);
+        return;
+    }
+
+    PlayBGM(no);
+}
+
+static void Snd_SetFieldDataWithoutDelay(u16 no, u16 unused) {
+    u16* zone_bgm = GF_SdatGetAttrPtr(SOUND_WORK_ZONE_BGM);
+    u32 tmp_bank_no = Snd_GetBank(*zone_bgm);   
+
+    if (tmp_bank_no == BANK_BASIC) {
+        GF_Snd_LoadSeqEx(no, NNS_SND_ARC_LOAD_WAVE);   
+        GF_ASSERT(FALSE);
+    }
+    else {
+        GF_Snd_LoadSeqEx(*zone_bgm, NNS_SND_ARC_LOAD_WAVE | NNS_SND_ARC_LOAD_BANK);
+    }
+
+    GF_Snd_SaveState(GF_SdatGetAttrPtr(SOUND_WORK_HEAP_SAVE_BGM));
+    
+    SndPause(PLAYER_FIELD, FALSE);               
+    GF_SndStartFadeInBGM(VOLUME_MAX, 40, VOLUME_MIN);  
+
+    Snd_SetBankFlag(SND_BANK_FLAG_CONTINUE);          
+}  
+
+void Snd_SetFieldSubData(u16 no, u16 old_bank_no) {
+    u32 tmp_bank_no;
+    u8* bank_flag    = GF_SdatGetAttrPtr(SOUND_WORK_BANK_FLAG);
+    u16* zone_bgm    = GF_SdatGetAttrPtr(SOUND_WORK_ZONE_BGM);
```

Comment:
Bad spacing, please sanitize

```suggestion
    u8 *bank_flag = GF_SdatGetAttrPtr(SOUND_WORK_BANK_FLAG);
    u16 *zone_bgm = GF_SdatGetAttrPtr(SOUND_WORK_ZONE_BGM);
```